### PR TITLE
style: Turn the CSS flexible box model on by default.

### PR DESCRIPTION
**Pull request by pcwalton**
See
https://groups.google.com/forum/#!topic/mozilla.dev.servo/MWBms2_BMJo
for discussion.

r? @larsbergstrom

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13590)

### CODE REVIEW

**Comment by stshine** [comment]
@bors-servo try
Since column flexbox is still not implemented there may be some false-positives. But that may not be a big deal, I suppose.

**Comment by stshine** [comment]
I am comfused that with prefs enabled on my local machine there are tens of tests passed. And how the current passed test been affected?
cc @SimonSapin

**Comment by larsbergstrom** [comment]
I'm in favor of this (and the code changes look good), provided that the changes in test expectations are reasonable. Flexbox is definitely starting to look better, and having it on by default will let us get more feedback, which is great while we also have multiple people actively working on it (especially @stshine and @shinglyu).

**Comment by stshine** [comment]
Sorry, did not notice the mailing list discussion before.
My original plan is to enable them after I have implemented column flexbox. But I fell into the rabbit hole of marking the flex items approperately during construction, and have been struggling in the mud of `nsCSSFrameConstructor.cpp` since then.
Since I may have digged too deep, IMHO it is generally good if you agree to enable them for practical reason if the number of false-positive tests enabled is small. And of course I hope to enable them, why not :smile:

**Comment by pcwalton** [comment]
@stshine I'm not sure what you mean by "I am comfused that with prefs enabled on my local machine there are tens of tests passed." Could you rephrase?

**Comment by stshine** [comment]
@pcwalton If run the tests with flex related properties enabled in `prefs.json` there will be more than ten additional tests got passed.

**Comment by pcwalton** [comment]
@bors-servo: try

I don't understand how the above test could possibly be related and it passes locally.

**Comment by pcwalton** [comment]
@stshine With this patch, the same number of flexbox tests pass as with master. Looking at one of them, I think margin handling is making a whole bunch of them fail.

This is a good reason to land this IMO: we can't catch regressions without it.

**Comment by stshine** [comment]
`flexbox_absolute-atomic.htm` seems to be the only false-positives since it has a abs-pos child in the flex container and the current code does nothing about them. Otherwise this should be OK.

**Comment by stshine** [comment]
See https://github.com/servo/servo/issues/12453#issuecomment-239611493 for a incomplete list of unimplemented features of flexbox if anyone is interested.

File: components/style/properties/longhand/box.mako.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@
         """.split()
         if product == "gecko":
             values += "-moz-box -moz-inline-box".split()
-        experimental_values = set("flex".split())
+        experimental_values = set()
```
Comment (emilio) [inline components/style/properties/longhand/box.mako.rs]: can you get rid of this variable entirely and its use below too?

**Review by emilio — COMMENTED** [review]
r=me with that if you want to get this landed. I think the consensus is that this is what most people wanted.

**Comment by jdm** [comment]
@pcwalton One small change required, then r+.

**Comment by stshine** [comment]
If this get merged it should close #12191, #12455 and #13576 .

**Comment by nox** [comment]
Superseded by #14111.
